src/utils.py

```diff
@@ -100,30 +100,3 @@ def batch_shape(self) -> torch.Size:
     @property
     def event_shape(self) -> torch.Size:
         return torch.Size([self.x_dim])
-    
-class RetrievalDistribution(dist.Distribution):
-    """A distribution that samples from a normalized gaussian and repeats a sampled x at the end of the prompt."""
-
-    def __init__(self, batch_shape: torch.Size, event_shape: torch.Size, *args: Any, **kwargs: Any):
-        super(RetrievalDistribution, self).__init__(*args, **kwargs | {"validate_args": False})
-        self.batch_size = batch_shape[0]
-        self.seq_len = batch_shape[1]
-        self.x_dim = event_shape[0]
-
-    def sample(self, query_pos: int = -1, sample_shape: torch.Size = torch.Size()):
-        is_random_query = query_pos < 0
-        x_batch = torch.randn(self.batch_size, 2 * self.seq_len + 1, self.x_dim)
-        for i in range(self.batch_size):
-            if is_random_query:
-                query_pos = 2 * int(torch.randint(0, self.seq_len, torch.Size()))
-            x_batch[i][-1, :] = x_batch[i][query_pos, :]
-        x_batch = x_batch / torch.norm(x_batch, p=2, dim=-1, keepdim=True) * math.sqrt(self.x_dim)
-        return x_batch
-
-    @property
-    def batch_shape(self) -> torch.Size:
-        return torch.Size([self.batch_size, self.seq_len])
-
-    @property
-    def event_shape(self) -> torch.Size:
-        return torch.Size([self.x_dim])
```

Comment:
removing RetrievalDistribution is fine, but something will have to be replaced for the x_dist. Right now, in `tasks.yml`, the type for x_dist is retrieval. Since we no longer have this, we will need to replace it with something that gets keys sampled from normalized gaussian.

Reply:
Actually, this might suggest transformable distributions specified in yaml -- e.g. sums, products, affine transforms of, etc of distributions